app/models/user.rb

```diff
@@ -24,10 +24,16 @@ def pending_friends
   end
 
   def friend_requests
-    friendships.map { |friendship| friendship.user unless friendship.status }.compact
+    invitations.map { |invitation| invitation.inviter unless invitation.status }.compact
   end
 
   def friend?(user)
     friends.include?(user)
   end
+
+  def accept_friend(inviterid)
```

Comment:
[OPTIONAL]
This could have also been done in the Friendship Model. In fact, it would help you avoid the query in line 36:
Something like following in Friendship Model
```ruby
def accept_friend
  self.update_attributes(confirmed: true)
  Friendship.create!(friend_id: self.user_id,
                  user_id: self.friend_id,
                  confirmed: true)
end
```